core/src/test/java/com/sap/ai/sdk/core/CacheTest.java

```diff
@@ -134,11 +146,52 @@ void newDeploymentAfterReset() {
     cacheUnderTest.loadCache(client, "default");
     stubGPT4();
 
-    cacheUnderTest.getDeploymentIdByModel(client, "default", "gpt-4-32k");
+    final AiModel targetModel = createAiModel("gpt-4-32k", null);
+
+    cacheUnderTest.getDeploymentIdByModel(client, "default", targetModel);
     // 1 reset empty and 1 cache miss
     wireMockServer.verify(2, getRequestedFor(urlPathEqualTo("/v2/lm/deployments")));
 
-    cacheUnderTest.getDeploymentIdByModel(client, "default", "gpt-4-32k");
+    cacheUnderTest.getDeploymentIdByModel(client, "default", targetModel);
     wireMockServer.verify(2, getRequestedFor(urlPathEqualTo("/v2/lm/deployments")));
   }
+
+  @Test
+  public void isDeploymentOfModel() {
+    // Create a target model
+    final AiModel targetModel = createAiModel("gpt-4-32k", null);
+    final AiModel targetModelWithDifferentVersion = createAiModel("gpt-4-32k", "1.0");
+
+    // Create a deployment with a different model by version
+    final var model = Map.of("model", Map.of("name", "gpt-4-32k", "version", "latest"));
+    final var deployment =
+        AiDeployment.create()
+            .id("test-deployment")
+            .configurationId("test-configuration")
+            .status(AiDeploymentStatus.RUNNING)
+            .createdAt(OffsetDateTime.parse("2024-01-22T17:57:23+00:00"))
+            .modifiedAt(OffsetDateTime.parse("2024-02-08T08:41:23+00:00"));
+    deployment.setDetails(
+        AiDeploymentDetails.create().resources(AiResourcesDetails.create().backendDetails(model)));
```

Comment:
This test should fail because we use `backend_details` instead of `backendDetails`, see DeploymentCache.java:157
You used `backendDetails`

To make this test fail, check that specifically requiring latest also works:
```java
final AiModel GPT4VersionLatest = createAiModel("gpt-4-32k", "latest");
assertThat(DeploymentCache.isDeploymentOfModel(GPT4VersionLatest, deployment)).isTrue();
```

Reply:
Why should it fail? The SDK can handle both to make sure the fix doesn't break our code.

Reply:
Because the spec is broken, and he used the generated code instead of a custom field (or a wiremock stub)

Reply:
That's fine, they are fixing it (or have already fixed it), so using the generated field should be okay?

Reply:
I don't understand why you think the test should fail. But I will include the additional test and update the code to reflect that `backend_details` is the correct field.